src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -39,8 +44,10 @@ public TypeFactory(OutputLibrary library)
             {
                 InputTypeKind.AzureLocation => new CSharpType(typeof(AzureLocation), inputType.IsNullable),
                 InputTypeKind.Boolean => new CSharpType(typeof(bool), inputType.IsNullable),
-                InputTypeKind.BytesBase64Url => new CSharpType(typeof(byte[]), inputType.IsNullable),
-                InputTypeKind.Bytes => new CSharpType(typeof(byte[]), inputType.IsNullable),
+                InputTypeKind.BytesBase64Url when _isCadl is true => new CSharpType(typeof(BinaryData), inputType.IsNullable),
```

Comment:
I think we should just add another `InputTypeKind` - `InputTypeKind.BinaryData`.
Add it to `InputTypeKind` in autorest C# and and replace `Bytes` with `BinaryData` in `InputTypeKind.ts`